### DISCUSSION TITLE
comment je peut changé les paramètres du fichier train.py

### DISCUSSION THREAD

**Discussion by SiwarRziguiING3** [ultralytics/yolov5#8268] [Q&A]
bonsoir, je veux trouvé les paramètres similaire a yolov4 dans yolov5 tel que  :
1.  max batches
2. filters

**Comment by glenn-jocher** [comment]
@SiwarRziguiING3 see train.py argparser for training settings:
https://github.com/ultralytics/yolov5/blob/1156a323f8fc221d5c1a36c700e72e129419d187/train.py#L477-L521

Model architectures are defined in each model's yaml, i.e. YOLOv5s:
https://github.com/ultralytics/yolov5/blob/master/models/yolov5s.yaml

**Reply by SiwarRziguiING3** [reply]
@glenn-jocher mais je ne trouve pas le paramètre 'filter' pour le changer

**Reply by glenn-jocher** [reply]
I'm not familiar with a 'filter' parameter. If you want to resize models you can use the compound scaling constants here:
https://github.com/ultralytics/yolov5/blob/1156a323f8fc221d5c1a36c700e72e129419d187/models/yolov5s.yaml#L5-L6

**Reply by SiwarRziguiING3** [reply]
@glenn-jocher d'accord merci beaucoup
j'ai un autre question : 
J'ai formé ma base de données personnalisée avec la taille réelle des images:
!python train.py --rect --batch 16 --epochs 500 --data custom.yaml --cfg ./models/yolov5s.yaml --weights yolov5s.pt --cache --freeze 12
je veux avoir quel est le size d'image que je peut le mètre pour faire le test dans detect.py

**Reply by glenn-jocher** [reply]
@SiwarRziguiING3 default --img is 640 for all files (train.py, val.py, detect.py, etc.). If you train and 640 then you'll get best results running inference also at 640:
https://github.com/ultralytics/yolov5/blob/a2a1ed201d150343a4f9912d644be2b210206984/train.py#L485

**Reply by SiwarRziguiING3** [reply]
@glenn-jocher @glenn-jocher J'ai entraîné ma base de données personnalisée avec la taille réelle de l'image sans redimensionner "--rect" au lieu de "--img 640". je veux savoir img-zise quand je fais le detect.py

**Reply by glenn-jocher** [reply]
@SiwarRziguiING3 --rect uses --img to define the long side of each image.

**Reply by SiwarRziguiING3** [reply]
@glenn-jocher je pense que je fait une confusion . quel est la différence entre --rect --img 640
